### ISSUE TITLE
Create migration script to import contact metadata from .yml files

### ISSUE THREAD

**Issue by pkarman** [usdoj/foia.gov#23]
The 2015 yml file build is being updated in #15.

It should be the baseline data, imported into the new Drupal instance for agencies to start with.

- [ ] a migration script that pulls data from yml file to the Drupal data model

**Comment by pkarman** [comment]
Now that #15 is complete, the yml files are up to date with 2016 data.

**Comment by ba66e77** [comment]
This is blocked by #16

**Comment by malikkotob** [comment]
Updated the dependencies here to show that it is blocked by the more granular tasks of building the data model in Drupal - #40 & #41 (as opposed to #16). Note that #40 and #41 are dependent on #16.

**Comment by malikkotob** [comment]
Questions posed to Matt in slack (with answers).

1. In the yml data, there are some officer/liaison/service center items that have phone numbers, but no names listed. For some of these cases, however, the phone number is associated with a name elsewhere. Here's an example in the DOE data where the service center number doesn't show a name - https://github.com/18F/2015-foia/blob/master/contacts/data/DOE.yaml#L520. If you scroll up a bit, however, you can see that the same number is used for the FOIA officer where there is a name listed - https://github.com/18F/2015-foia/blob/master/contacts/data/DOE.yaml#L391. Rather than create two FOIA personnel entities in Drupal, one for the officer with the name and number, and a second for the service center with just the number and *no* name, would you prefer that I consolidate the two and use the information with both name and number for both the officer and personnel fields?

    > No, lets leave them separate - i know this might entail a little more effort, but some FOIA offices do it for a reason.  The idea being is that while these two distinct entities (Service Center and FOIA Officer) may share a common phone number, they have two distinct roles and responsibilities and keeping that delineation clean is helpful.

1. There are cases where officer/liaison/service center items again have phone numbers and no names listed, but this time there are no names listed elsewhere that are tied to that number. When they're migrated into Drupal they are added as FOIA Personnel entities with a phone number and no name. Would you like some sort of default name to be tied to these numbers? E.g. if it's a service center number with no name, use 'Service Center Contact', if it's an officer number with no name use 'Officer Contact', etc.?

    > Yes, if there is no name associated with a particular entity, it would be nice to have a "generic" descriptive name associated with the contact information.  Some small agencies and components do this because they may need more that one person to fill a role by splitting time to cover it, or there may be frequent turnover, etc...

**Comment by adborden** [comment]
@malikkotob maybe you were thinking this already, re (2), I would leave this as blank/null in the datastore, and any default display name should be handled on the front end as part of the display.

**Comment by malikkotob** [comment]
@adborden I was thinking that actually, but the only issue would be if a component wants to reuse that same FOIA personnel entity on another field, they wouldn't have a way to search that existing entity.

What's happening currently with this scenario is as follows:
1. An agency component shows a phone number and no name, for example, for the public liaison object in the source data
1. The migration scripts creates a FOIA Personnel entity with no name and a phone number
1. The public liaison field for the agency component references the newly created personnel entity

If, at some point, the component decides they'd like to go in and use that same personnel entity for the FOIA officer field by leveraging the 'Add existing FOIA personnel' feature, they won't be able to look that existing personnel entity up, because there will be no name to search it by. They will be forced then, to use the 'Add new FOIA personnel' feature for the officer field, and introduce duplicative data.

My thinking was that we could default the name to something generic on the backend to remedy this, to something like '<Insert Agency Component name here> contact' (or something better). This would allow them to edit the name should they choose to down the road but make it searchable in the mean time should they choose to leverage the same personnel across the different fields (officer/liaison/service center).

Thoughts?

**Comment by adborden** [comment]
I see, that makes sense but this is getting into the UX realm which we'll want to test first. I wouldn't spend much time on it until we can get more feedback from users. The initial wireframes for the contact updater are very simple, i.e. there is no person entity that the agency would interact with.
